pkg/ivd/ivd_protected_entity.go

```diff
@@ -115,8 +115,8 @@ func (this IVDProtectedEntity) getDataWriter(ctx context.Context) (io.WriteClose
 func (this IVDProtectedEntity) getDiskConnectionParams(ctx context.Context, readOnly bool) (gDiskLib.ConnectParams, error) {
 	url := this.ipetm.client.URL()
 	serverName := url.Hostname()
-	userName := this.ipetm.user
-	password := this.ipetm.password
+	userName := this.ipetm.vcParams["user"].(string)
```

Comment:
This has room for errors to creep in since the strings can get out of sync.  The original code would fail at compile time if the field name was changed without rippling things through.  If "user" changes again, then it needs to be updated everywhere.  Let's make a function that gets called to retrieve the user name.  Let's make the keys (user, password, VirtualCenter, etc) constants as well.

Reply:
Made the change based on your recommendation.